src/v/redpanda/application.cc

```diff
@@ -693,6 +693,7 @@ void application::wire_up_redpanda_services() {
       &_rm_group_proxy,
       std::ref(rm_partition_frontend))
       .get();
+    tx_gateway_frontend.local().rearm_expire_timer();
```

Comment:
this should be moved into an appropriate spot in tx_gateway_frontend (e.g. maybe setup/start/ctor phase)

Reply:
yeah, but tx_gateway_frontend is a sharded object, if I use ctor to start the timer than it will be active on all cores instead of one. i thought it's a lesser evil to start it once there than to introduce coordination between the cores

Reply:
in the ctor (or start method) you can check if you are on core-0, make other cores a noop, and write a comment explaining why it only runs on one core. since that only happens once at boot up, I think the self-containment on the frontend implementation outweighs the special case.